pkg/specgen/generate/storage.go

```diff
@@ -67,34 +71,54 @@ func finalizeMounts(ctx context.Context, s *specgen.SpecGenerator, rt *libpod.Ru
 	}
 
 	for _, m := range commonMounts {
-		if _, ok := unifiedMounts[m.Destination]; !ok {
-			unifiedMounts[m.Destination] = m
+		if err = parse.ValidateVolumeCtrDir(m.Destination); err != nil {
+			return nil, nil, nil, errors.Wrapf(err, "validating destination path for common mount %q", m)
```

Comment:
The output of this error is going to be hard to interpret by the user?  The m struct for the user is going to be ugly. Does the user need it, and does the user know what a "common mount" is?